pkg/cmd/server/api/types.go

```diff
@@ -117,6 +117,11 @@ type MasterConfig struct {
 	// PauseControllers instructs the master to not automatically start controllers, but instead
 	// to wait until a notification to the server is received before launching them.
 	PauseControllers bool
+	// ControllerLeaseTTL enables controller election, instructing the master to attempt to acquire
+	// a lease before controllers start and renewing it within a number of seconds defined by this value.
+	// Setting this value non-negative forces pauseControllers=true. This value defaults off (0, or
+	// omitted) and controller election can be disabled with -1.
+	ControllerLeaseTTL int
```

Comment:
Then we have to do backwards compat and conversions.  I'd like to do that
on the next field we add, I'll add a TODO in there for it.

On Fri, Aug 14, 2015 at 3:56 PM, David Eads <notifications@github.com>
wrote:

> In pkg/cmd/server/api/types.go
> <https://github.com/openshift/origin/pull/4137#discussion_r37113261>:
>
> > @@ -117,6 +117,11 @@ type MasterConfig struct {
> >  	// PauseControllers instructs the master to not automatically start controllers, but instead
> >  	// to wait until a notification to the server is received before launching them.
> >  	PauseControllers bool
> > +	// ControllerLeaseTTL enables controller election, instructing the master to attempt to acquire
> > +	// a lease before controllers start and renewing it within a number of seconds defined by this value.
> > +	// Setting this value non-negative forces pauseControllers=true. This value defaults off (0, or
> > +	// omitted) and controller election can be disabled with -1.
> > +	ControllerLeaseTTL int
>
> Third part related to controllers. I smell a new struct.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/4137/files#r37113261>.
>



-- 
Clayton Coleman | Lead Engineer, OpenShift